Adafruit_FRAM_SPI.cpp

```diff
@@ -1,89 +1,110 @@
+  } ///< Post-SPI
 
-/**************************************************************************/
 /*!
-    Constructor
-*/
-/**************************************************************************/
-Adafruit_FRAM_SPI::Adafruit_FRAM_SPI()
-{
-  _cs = _clk = _mosi = _miso = -1;
-  _framInitialised = false;
-}
-Adafruit_FRAM_SPI::Adafruit_FRAM_SPI(int8_t cs)
-{
+ *  @brief  Instantiates a new SPI FRAM class using hardware SPI
+ *  @param  cs
+ *          optional cs pin number
+ *  @param  *theSPI
+ *          SPI optional object
+ */
+Adafruit_FRAM_SPI::Adafruit_FRAM_SPI(int8_t cs, SPIClass *theSPI = &SPI) {
   _cs = cs;
   _clk = _mosi = _miso = -1;
   _framInitialised = false;
+  _spi = theSPI;
 }
 
-Adafruit_FRAM_SPI::Adafruit_FRAM_SPI(int8_t clk, int8_t miso, int8_t mosi, int8_t cs)
-{
-  _cs = cs; _clk = clk; _mosi = mosi; _miso = miso;
+/*!
+ *  @brief  Instantiates a new SPI FRAM class using bitbang SPI
+ *  @param  clk
+ *          CLK pin
+ *  @param  miso
+ *          MISO pin
+ *  @param  mosi
+ *          MOSI pin
+ *  @param  cs
+ *          CS pin
+ */
+Adafruit_FRAM_SPI::Adafruit_FRAM_SPI(int8_t clk, int8_t miso, int8_t mosi,
+                                     int8_t cs) {
+  _cs = cs;
+  _clk = clk;
+  _mosi = mosi;
+  _miso = miso;
   _framInitialised = false;
 }
 
-/*========================================================================*/
-/*                           PUBLIC FUNCTIONS                             */
-/*========================================================================*/
-/**************************************************************************/
 /*!
-    Initializes SPI and configures the chip (call this function before
-    doing anything else)
-*/
-/**************************************************************************/
-boolean Adafruit_FRAM_SPI::begin()
-{
-	return begin(_cs, 2);
-}
+ *  @brief  Initializes SPI and configures the chip (call this function before
+ *          doing anything else)
+ *  @return true if succesful
+ */
+boolean Adafruit_FRAM_SPI::begin() { return begin(_cs, 2); }
 
-boolean Adafruit_FRAM_SPI::begin(uint8_t nAddressSizeBytes)
-{
-	return begin(_cs, nAddressSizeBytes);
+/*!
+ *  @brief  Initializes SPI and configures the chip (call this function before
+ *          doing anything else)
+ *  @param  nAddressSizeBytes
+ *          address size in bytes
+ *  @return true if succesful
+ */
+boolean Adafruit_FRAM_SPI::begin(uint8_t nAddressSizeBytes) {
+  return begin(_cs, nAddressSizeBytes);
 }
 
-boolean Adafruit_FRAM_SPI::begin(int8_t cs, uint8_t nAddressSizeBytes)
-{
-  if (cs == -1)
-  {
-    //Serial.println("No cs pin specified!");
+/*!
+ *  @brief  Initializes SPI and configures the chip (call this function before
+ *          doing anything else)
+ *  @param  cs
+ *          cs pin num
+ *  @param  nAddressSizeBytes
+ *          sddress size in bytes
+ *  @return true if succesful
+ */
+boolean Adafruit_FRAM_SPI::begin(int8_t cs, uint8_t nAddressSizeBytes) {
```

Comment:
change this to use the _cs we already defined, remove the argument here and default naddresize to 2